integrations/opensearch/tests/test_document_store.py

```diff
@@ -96,6 +112,153 @@ def test_get_default_mappings(_mock_opensearch_client):
     }
 
 
+class TestAuth:
+    @pytest.fixture(autouse=True)
+    def mock_boto3_session(self):
+        with patch("boto3.Session") as mock_client:
+            yield mock_client
+
+    @patch("haystack_integrations.document_stores.opensearch.document_store.OpenSearch")
+    def test_init_with_basic_auth(self, _mock_opensearch_client):
+        OpenSearchDocumentStore(hosts="testhost", http_auth=("user", "pw")).client
```

Comment:
The access to the `client` property is a bit hidden, I would change this and the other tests to something similar to this:

```
document_store = OpenSearchDocumentStore(hosts="testhost", http_auth=("user", "pw"))
assert document_store.client
```

At that point you can also remove `# ruff: noqa: B018` from above I think.